net: IPv6: Fix the wrong source address during mesh multicasting

lmaciejonczyk opened the pull request.

jukkar commented on subsys/net/ip/net_if.c:
```diff
@@ -2347,7 +2347,8 @@ static struct in6_addr *net_if_ipv6_get_best_match(struct net_if *iface,
 			 */
-			if (ipv6->unicast[i].is_mesh_local && len < 64) {
+			if (ipv6->unicast[i].is_mesh_local &&
+			    (len < 64 && !net_ipv6_is_addr_mcast_mesh(dst))) {
```
No need to add `()` here

jukkar reviewed (approved):
LGTM

lmaciejonczyk committed:
net: IPv6: Fix source address for mesh multicast destination

Currently there is chosen the link local address as a source address
for each multicast destination address. It is a bug for OpenThread
network where the mesh-local EID addres should be picked in this case.
This commit fixes it by distinquish the mesh local multicast among any
others.

Signed-off-by: Lukasz Maciejonczyk <lukasz.maciejonczyk@nordicsemi.no>

lmaciejonczyk committed:
tests: net: Add tests for IPv6 mesh addresses

This tests make sure that the right source address is chosen during
the mesh multicasting.

Signed-off-by: Lukasz Maciejonczyk <lukasz.maciejonczyk@nordicsemi.no>

rlubos reviewed (approved):
LGTM, thanks!
